docs/spec_draft.md

```diff
@@ -1029,6 +1030,52 @@ specification. Numeric precision is implementation-defined.
 
 [Back to Ops](#index-of-documented-ops)
 
+## stablehlo.transpose
+
+`stablehlo.transpose(operand, permutation) -> result`
+
+### Semantics
+
+Permutes the dimensions of `operand` tensor using a permutation of its shape
+`permutation` and produces a `result` tensor. The $i^{th}$ dimension of the
+`result` is given by `permutation`$[i]^{th}$ dimension of the `operand`.
+
+### Operands
+
+| Name | Type |
+|-|-|
+| `operand` | tensor of any supported types |
+| `permutation` | array of type `ui64` |
```

Comment:
For better or for worse, this is currently an array of `si64`, so let's start with reflecting the reality, and in the future we'll be able to change this to `ui64` if we decide.